FWCore/ParameterSet/python/Config.py

```diff
@@ -2635,6 +2702,36 @@ def testPath(self):
             t = Path(p.a, p.t1, Task(), p.t1)
             self.assertTrue(t.dumpPython(PrintOptions()) == 'cms.Path(process.a, cms.Task(), process.t1)\n')
 
+        def testFinalPath(self):
+            p = Process("test")
+            p.a = OutputModule("MyOutputModule")
+            p.b = OutputModule("YourOutputModule")
+            p.c = OutputModule("OurOutputModule")
+            path = FinalPath(p.a)
+            path *= p.b
+            path += p.c
+            self.assertEqual(str(path),'a+b+c')
+            path = FinalPath(p.a*p.b+p.c)
+            self.assertEqual(str(path),'a+b+c')
+            path = FinalPath(p.a+ p.b*p.c)
+            self.assertEqual(str(path),'a+b+c')
+            path = FinalPath(p.a*(p.b+p.c))
+            self.assertEqual(str(path),'a+b+c')
+            p.es = ESProducer("AnESProducer")
+            self.assertRaises(TypeError,Path,p.es)
```

Comment:
Should this test
```suggestion
            self.assertRaises(TypeError,FinalPath,p.es)
```
instead?

Reply:
fixed